server/index-node/src/resolver.rs

```diff
@@ -34,19 +35,41 @@ impl TryFromValue for DeploymentAssignment {
     }
 }
 
+#[derive(Debug)]
+struct ProofOfIndexingInput {
+    // The id of the subgraph.
+    subgraph_name: String,
+    // The block number in ethereum to query against. Eventually when the data
+    // availability chain comes online and we support other block chains then
+    // this will be come an optional field.
+    ethereum_block_number: BigInt,
```

Comment:
IMHO this should be just `block` or `block_number` (with the comment reflecting that it's the block number on the chain that the subgraph is indexing, which could be an Ethereum chain). The reason I'd like to get rid of the `ethereum` here is that this will work just as well for other blockchains.

Reply:
We had started out with an API in the RFC that was something like `blocks: [Block!]!` where `Block` was an interface and there was `EthereumBlock` etc. That doesn't work, because GraphQL doesn't (yet?) support interfaces in inputs. We still need to be able to support blocks from multiple blockchains in the same query - especially with the availability chain coming.

So, there's a couple of APIs possible. One, you could keep in the spirit of the per/network blocks without the interface, but then it's easy to specify more than one block with the same network. Eg: `[{Block {network: "eth", number: 10}, Block {network: "eth", number: 20}}]` It's unclear what to return (probably an error, even though the data is not malformed). With the current API, this isn't a problem and has the same amount of API churn as the original `EthereumBlock` API.

I'm not really happy with either API really. Maybe we can defer until we actually have multiple chains?